source/YTEPs/YTEP-0037.rst

```diff
@@ -0,0 +1,261 @@
+that goes beyond code style and detects some additional anti-patterns, most of which are
+correspond very likely to design flaws in otherwise syntaxically valid statement. For
+instance, it will catch mutable default values such as in
+
+.. code:: python
+
+    def spam(a={}, b=[]):
+        #...
+
+which, in most contexts, should be rewritten as 
+
+.. code:: python
+
+    def spam(a=None, b=None):
+        if a is None:
+            a = {}
+        if b is None:
+            b = []
+        
+This is a well known "gotcha", as documented for instance
+`here <https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments>`_.
+In short, this plugin detects bugs that went under the radar up to now, so it's
+probably worth adding it to our linting CI.
+
+--
+
+Another plugging can be added to enforce docstring formatting
+(`flake8-docstrings <https://github.com/PyCQA/flake8-docstrings>`_), and has a
+straight-forward option configuration to validate docstrings are numpy-styled. However,
+there is currently a very large debt in errors caught by this tool, and no way to
+automatically solve them. However, it could still be added to our linting CI, if check
+for *new* errors only, such as
+
+.. code:: bash
+
+    git diff upstream/master -u -- "*.py" | flake8 --diff
+
+(snippet borrowed from pandas' contributing guide)
+
+
+
+**side effects**
+
+Although some default options in ``isort`` conflict with ``black``'s opinated standard,
+it can be configured so that the tools play nicely with each other.
+This is demonstrated in `#2596 <https://github.com/yt-project/yt/pull/2596>`_ where both
+check pass on Travis.
+
+On another note, black only recognizes ``pyproject.toml`` as a configuration file (and
+is explicitly not planning to support other files such as ``setup.cfg``).
+An undesirable effect of using  ``pyproject.toml`` solely as a configuration file for
+``black`` is that ``pip`` will detect it and change its behaviour when its present. The
+correct way to introduce this file is by specifying yt's build requirements within it. 
+A proof of concept for this is `#2598 <https://github.com/yt-project/yt/pull/2598>`_,
+where CI builds are run correctly across all tested python versions (3.6, 3.7, 3.8).
+
+A serious counter-argument to applying black is that it implies messing up with ``git
+blame`` by making a single contributor the defacto last-author of a large number of
+lines they have not even necessarily read. Most recent versions of ``git`` can be
+configured to ignore specific commits in ``git blame``. However, ``black``'s own README
+currently points out that github's UI for ``git-blame`` does not support this feature
+(yet ?).
+
+It should be noted that ``black`` does not have a parser for Cython files, but
+interstingly ``flake8`` and ``isort`` do. Thus it is possible to add style checks for
+Cython extensions to the CI pipeline.
+
+Additionally, ``black`` will not force line-lenght limits in docstrings.
```

Comment:
line-lenght --> line-length

Reply:
oops, not the only time I've misspelled this apparently. Thanks !